### PR TITLE
Improve ExpressionBuilder test coverage

### LINKED ISSUE #6448: Improve ExpressionBuilder unit test
improve test coverage

### PR DESCRIPTION
**Pull request by dschwen**
Add tests for substitutions and other subtleties.

Closes #6448

### CODE REVIEW

File: unit/src/ExpressionBuilderTest.C [hunk 1]
```diff
@@ -34,45 +35,95 @@ void ExpressionBuilderTest::test()
...
   // Test using functions in terms
-  G((x,y)) = f;
-  H((x)) = -G((x,b)) + c % b;
+  {
+    G((x,y)) = f;
+    H((x)) = -G((x,b)) + c % b;
 
-  CPPUNIT_ASSERT( std::string(H) == "-(x^2+4^1+3*4*-1)+8.9%4" );
-  CPPUNIT_ASSERT( G.args() == "x,y" );
+    CPPUNIT_ASSERT( std::string(H) == "-(x^2+4^1+3*4*-1)+8.9%4" );
```
Comment (permcody) [inline unit/src/ExpressionBuilderTest.C]: Is that a unicode character or just something displaying oddly in my browser?

File: unit/src/ExpressionBuilderTest.C [hunk 2]
```diff
@@ -34,45 +35,95 @@ void ExpressionBuilderTest::test()
...
+  {
+    // plog substitution
+    EBTerm u = log(x/a);
+    u.substitute(EBLogPlogSubstitution(b));
+    CPPUNIT_ASSERT( std::string(u) == "plog(x/1,4)" );
+
+    // single substitution in a ternary
+    EBTerm v = conditional(x < y, x * y, x / y);
+    v.substitute(EBTermSubstitution(x, a));
+    CPPUNIT_ASSERT( std::string(v) == "if(1<y,1*y,1/y)" );
```
Comment (permcody) [inline unit/src/ExpressionBuilderTest.C]: This is probably the reason for your failed pre-check.

Comment (dschwen) [inline unit/src/ExpressionBuilderTest.C]: Oh right, I've had that before...

File: unit/src/ExpressionBuilderTest.C (hunk 1 above)
Comment (dschwen) [inline unit/src/ExpressionBuilderTest.C]: It is just GitHub's syntax highlighting f*ing up a simple modulo operator

**Pull request merged**